src/NexusMods.Paths/FileTree/FileTreeNode.cs

```diff
@@ -201,6 +212,57 @@ public void Deconstruct(out TPath path, out TValue? value)
         value = Value;
     }
 
+
+    /// <summary>
+    /// For a given subpath, find all nodes that match the subpath. The returned nodes will be the top of the subpath.
+    /// This allows for searching `/foo/bar/baz/qux` by passing `bar/baz` to this function and this function will then
+    /// return the `bar` node.
+    /// </summary>
+    /// <param name="subpath"></param>
+    /// <returns></returns>
+    public IEnumerable<FileTreeNode<TPath, TValue>> FindSubPath(RelativePath subpath)
+    {
+        var parts = subpath.Parts.ToArray();
+
+        var children = GetAllNodes().Where(n => n.Path.StartsWith(parts[0]));
+
+        foreach (var (_, childNode) in children)
+        {
+            var notFound = false;
+            var node = childNode;
+            foreach (var part in parts[1..])
+            {
+                if (!node.Children.TryGetValue(part, out var found))
+                {
+                    notFound = true;
+                    break;
+                }
+
+                node = found;
+            }
+
+            if (notFound) continue;
+            yield return childNode;
+        }
+    }
+
+    /// <summary>
+    /// Gets all nodes in the tree.
+    /// </summary>
+    /// <returns></returns>
+    public IEnumerable<(RelativePath Path, FileTreeNode<TPath, TValue> Node)> GetAllNodes()
```

Comment:
Please change Path to either `RelatPath` or `Relative` or something to not make it overload with the Node.Path property, which is a TPath and relative to the Root of the node.
I was confused for a while here: https://github.com/Nexus-Mods/NexusMods.Paths/pull/14/files#diff-8549218ef2c68050733b33a11127691c206b485ac8609f508898818164154cabR227
as I was assuming that `n.Path` was `TreeNode.Path`  rather than the named tuple output of GetAllNodes().